### PR TITLE
PROD-86: fix email confirmation and views

### PR DESCRIPTION
**Pull request by anastasiabalaban**
# Reviewers

- [x] @vikabelous
- [x] @okohuch
- [x] @shevchukm 
- [x] @Pavliv

# Jira task

http://ssu-jira.softserveinc.com/browse/PROD-86

# Description

- add asserts to public folder
- add necessary configs to production
- change views with gem simple_form
- add flash messages

# Checklist:

- [x] Rubocop doesn't raise any warnings and offenses
- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

File: app/views/devise/passwords/edit.html.haml [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
+      = simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put, novalidate: false }) do |f|
         = f.hidden_field :reset_password_token
         .form-group
           = f.label :password, t('.new_password')
-          - if @minimum_password_length
-            %em= t('devise.shared.minimum_password_length', count: @minimum_password_length)
-            %br
-          = f.password_field :password, autofocus: true, autocomplete: "off", class: "form-control"
+          %em= t('devise.shared.minimum_password_length', count: @minimum_password_length)
+          = f.input :password, label: t('.new_password'), autofocus: true, autocomplete: "off", input_html: { class: 'form-control', minlength: 6 }, label: false
```
Comment (Pavliv) [inline app/views/devise/passwords/edit.html.haml]: Use ' ' instead " "

File: app/views/devise/passwords/edit.html.haml [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
         .form-group
-          = f.label :password_confirmation, t('.confirm_new_password')
-          %br
-          = f.password_field :password_confirmation, autocomplete: "off", class: "form-control"
+          = f.input :password_confirmation, label: t('.confirm_new_password'), autocomplete: "off", input_html: { class: 'form-control' }
```
Comment (Pavliv) [inline app/views/devise/passwords/edit.html.haml]: The same here

File: app/views/devise/passwords/edit.html.haml [hunk 3]
```diff
@@ -1,20 +1,15 @@
...
         .actions
-          = f.submit t('.change_my_password'), class: "btn btn-outline-info"
+          = f.submit t('.change_my_password'), class: "btn btn-outline-info btn-size"
```
Comment (Pavliv) [inline app/views/devise/passwords/edit.html.haml]: And here

File: app/views/devise/shared/_links.html.haml [hunk 4]
```diff
@@ -1,19 +1,19 @@
...
+  = link_to t('.didn_t_receive_confirmation_instructions'), new_confirmation_path(resource_name), class: "link"
   %br
 - if devise_mapping.lockable? && resource_class.unlock_strategy_enabled?(:email) && controller_name != 'unlocks'
-  = link_to t('.didn_t_receive_unlock_instructions'), new_unlock_path(resource_name)
+  = link_to t('.didn_t_receive_unlock_instructions'), new_unlock_path(resource_name), class: "link"
   %br
 - if devise_mapping.omniauthable?
   - resource_class.omniauth_providers.each do |provider|
-    = link_to t('.sign_in_with_provider', provider: OmniAuth::Utils.camelize(provider)), omniauth_authorize_path(resource_name, provider)
+    = link_to t('.sign_in_with_provider', provider: OmniAuth::Utils.camelize(provider)), omniauth_authorize_path(resource_name, provider), class: "link"
```
Comment (Pavliv) [inline app/views/devise/shared/_links.html.haml]: Replace " " to ' '

File: app/views/layouts/application.html.haml [hunk 5]
```diff
@@ -53,6 +53,10 @@
                 %a.nav-link{href: "#"}= t(".contacts")
     %section.no-gutters
       .container
+        - flash.each do |message_type, message|
+          .alert.alert-info
+            %button.close{"data-dismiss" => "alert", type: "button"} ×
+            #{message}
```
Comment (Pavliv) [inline app/views/layouts/application.html.haml]: The same

**Review by Pavliv — APPROVED** [review]

**Review by shevchukm — APPROVED** [review]

File: config/initializers/devise.rb [hunk 6]
```diff
@@ -62,4 +62,6 @@
   config.sign_out_via = :delete
 
   config.mailer = 'DeviseCustomMailer'
+
+  config.secret_key = '9c72504bb7fca71b425dc1b89c429f405643ab21e035315b42330daadcf1a47fd5fd3583971bd6560c5be934ece2d7fbae7343b58d56f4c72a7369e934573132'
```
Comment (viktoriia-bilous) [inline config/initializers/devise.rb]: This value should be stored as ENV variable as well

File: config/initializers/simple_form.rb [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+    # Calculates readonly automatically from readonly attributes
+    b.optional :readonly
+
+    ## Inputs
+    # b.use :input, class: 'input', error_class: 'is-invalid', valid_class: 'is-valid'
+    b.use :label_input
+    b.use :hint,  wrap_with: { tag: :span, class: :hint }
+    b.use :error, wrap_with: { tag: :span, class: :error }
+
+    ## full_messages_for
```
Comment (viktoriia-bilous) [inline config/initializers/simple_form.rb]: Please review this file and either remove unused settings with their comments or uncomment those settings you would like to use.

File: public/assets/.sprockets-manifest-413d41b33cf339d3ef4b4c35f32e8402.json [hunk 8]
```diff
@@ -0,0 +1 @@
...st":"875cca143a022ebb6768c8901324671170a0eaceaa9e3d4969cb2f4f71e3e2b7","integrity":"sha256-h1zKFDoCLrtnaMiQEyRnEXCg6s6qnj1JacsvT3Hj4rc="},"application-b1b11a83199bd0be2335bfd4ac6c28ce9b7a64da1ec0d2af3b73b67fceeead78.css":{"logical_path":"application.css","mtime":"2018-04-16T15:59:06+03:00","size":126325,"digest":"b1b11a83199bd0be2335bfd4ac6c28ce9b7a64da1ec0d2af3b73b67fceeead78","integrity":"sha256-sbEagxmb0L4jNb/UrGwozpt6ZNoewNKvO3O2f87urXg="}},"assets":{"email.css":"email-d6b8be8bae4f838685f2f88e112702f4fef0e9c169d4ca25d52bcb14fdf7bb3b.css","edit.png":"edit-72a5ccd30f62c136fe038aea011169f6503d7f2799d2874fda91f541642bf664.png","hand.png":"hand-d0394b3f15228194d1f7338a314f909175699c021f48ae2912b6c4eee45ae48a.png","ok.png":"ok-bc1d6c587bbe12cd805c455b607075d14d7dbd70b02fd24bf2ceddfd447d575d.png","prodobro1.png":"prodobro1-7eb2a1e25ade1b4785a7a1d98eb72810bb493acae4735d7b2d75dae9ff7331b0.png","prodobro2.png":"prodobro2-9d9dcdf002f446dd3b77ccea0091ee1f75713ad69bdc8288e23a41834dfe29c2.png","application.js":"application-875cca143a022ebb6768c8901324671170a0eaceaa9e3d4969cb2f4f71e3e2b7.js","application.css":"application-b1b11a83199bd0be2335bfd4ac6c28ce9b7a64da1ec0d2af3b73b67fceeead78.css"}}
```
Comment (viktoriia-bilous) [inline public/assets/.sprockets-manifest-413d41b33cf339d3ef4b4c35f32e8402.json]: `public/assets` folder should be in` .gitignor`
Assets should be compiled during the deploy process

**Review by viktoriia-bilous — CHANGES_REQUESTED** [review]

File: config/locales/simple_form.en.yml [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    "no": 'No'
+    required:
+      text: 'required'
+      mark: '*'
+      # You can uncomment the line below if you need to overwrite the whole required html.
+      # When using html, text and mark won't be used.
+      # html: '<abbr title="required">*</abbr>'
+    error_notification:
+      default_message: "Please review the problems below:"
+    # Examples
```
Comment (viktoriia-bilous) [inline config/locales/simple_form.en.yml]: Please delete unnecessary comments

**Review by okohuch — APPROVED** [review]

File: app/views/devise/passwords/edit.html.haml [hunk 10]
```diff
@@ -1,20 +1,15 @@
...
+      = simple_form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put, novalidate: false }) do |f|
         = f.hidden_field :reset_password_token
         .form-group
           = f.label :password, t('.new_password')
-          - if @minimum_password_length
-            %em= t('devise.shared.minimum_password_length', count: @minimum_password_length)
-            %br
-          = f.password_field :password, autofocus: true, autocomplete: "off", class: "form-control"
+          %em= t('devise.shared.minimum_password_length', count: @minimum_password_length)
+          = f.input :password, label: t('.new_password'), autofocus: true, autocomplete: 'off', input_html: { class: 'form-control', id: 'password1', minlength: 6 }, label: false
```
Comment (viktoriia-bilous) [inline app/views/devise/passwords/edit.html.haml]: `label: t('.new_password')`
`label: false`

File: app/assets/javascripts/users.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+window.onload = function () {
```
Comment (viktoriia-bilous) [inline app/assets/javascripts/users.js]: You need a tool like rubocop for js files. This file has some style issues. Consider any of these gems:
https://github.com/appfolio/eslint-rails
https://github.com/damian/jshint
https://github.com/mackuba/jslint_on_rails

Add a chosen gem, configure it, run, fix errors, add to CI script as well.

**Review by viktoriia-bilous — CHANGES_REQUESTED** [review]

**Review by viktoriia-bilous — APPROVED** [review]
Good job :)

**Pull request merged**